fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -650,8 +661,10 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
     public MultiResourceResult<Resource> search(FHIRPersistenceContext context, Class<? extends Resource> resourceType)
             throws FHIRPersistenceException {
 
-        // Fall back to the old search code if new query builder has been disabled.
-        if (!this.optQueryBuilderEnabled) {
+        // Fall back to the old search code if new query builder has been disabled
+        // or if whole-system search and legacy whole-system search params enabled.
+        if (!this.optQueryBuilderEnabled ||
+                (this.legacyWholeSystemSearchParamsEnabled && isSystemLevelSearch(resourceType))) {
             log.warning("The server is configured to use the legacy query builder via fhirServer/search/enableOptQueryBuilder."
```

Comment:
Update the warning message since it might be because of legacyWholeSystemSearchParamsEnabled instead of enableOptQueryBuilder.

Reply:
Updated.